### PR TITLE
Update the docker/spdystream dependency

### LINKED ISSUE #19230: unit flake: pkg/client/unversioned/portforward is timing out
FAIL	k8s.io/kubernetes/pkg/client/unversioned/portforward	300.229s

Example failure:
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/e41d66f0e207f8d32268b80e8d0d1c227663cc48/kubernetes-pull-test-unit-integration/7816/artifacts/?debugUI=DEVELOPERS

### PR DESCRIPTION
**Pull request by brendandburns**
According to @ncdc this may fix #19230 via https://github.com/docker/spdystream/pull/64

@ixdy @spxtr

### CODE REVIEW

**Comment by ncdc** [comment]
@brendandburns thanks, you beat me to it :smile:

It looks like there's an issue verifying the godep bump in travis... Let me know if you need help figuring out what's up.

**Comment by thockin** [comment]
Should we accelerate this to try to get the test flake killed?

**Comment by thockin** [comment]
re: godep test fail, I think you maybe need to update your godep version, Brendan.  ISTR something in Godep changed here in the last few months.

**Comment by brendandburns** [comment]
ok, godep version updated, let's hope this passes godep validation.

File: Godeps/LICENSES.md [hunk 1]
```diff
@@ -29,7 +29,8 @@ github.com/daviddengcn/go-colortext | BSD?
 github.com/dgrijalva/jwt-go | spdxMIT
 github.com/docker/docker | Apache-2
 github.com/docker/docker/pkg/symlink | spdxBSD3
-github.com/docker/spdystream | Apache-2
+github.com/docker/spdystream | SeeFile
+. |
```
Comment (lavalamp) [inline Godeps/LICENSES.md]: @karlkfi is this output you expect in this file?

**Comment by lavalamp** [comment]
lgtm too as soon as everything is happy.

**Comment by brendandburns** [comment]
I re-ran `./hack/update-godep-licenses.sh` on my box and the output is unchanged.  I think that the SPDY stream license actually has changed with this import of the new LICENSE.docs file.

@karlkfi can you help debug the problem here?

**Comment by brendandburns** [comment]
@lavalamp I'm tempted to just merge this.  I _think_ that the error is a bug in the validate rather than an actual problem.  There really are a bunch of licenses included now, so I think that the change is right, I think that the test might just not expect that a license could change.

**Comment by lavalamp** [comment]
I'm fine with that as long as it doesn't leave tests red :)

**Comment by lavalamp** [comment]
You could also try adding an exception-- there's examples in the script that generates that file.

**Comment by thockin** [comment]
The second commit in this series does change the license for spdystream:

```
 +github.com/docker/spdystream | SeeFile
 +. | 
```

That doesn't seem right on its face - the . is clearly wrong.  Running the updater (from mesos - @karlkfi for a clue?) gives me Apache-2, but throws in some other blank lines.

try running `hack/update-godep-licenses.sh` again?

**Comment by brendandburns** [comment]
@thockin note that this PR adds a new file `LICENSE.docs` which includes a large number of different licenses.  I suspect this is why it now says `SeeFile`

When you ran it on your machine, did you patch in this PR?  Without this PR, I would expect `Apache-2` since that's what the other license contains.

I have re-run this on my workstation and it continues to generate `SeeFile`

--brendan

**Comment by brendandburns** [comment]
@lavalamp I added an exception and this is passing now... (hopefully travis agrees)

**Comment by thockin** [comment]
It's not clear whether thsi is just an exceptional case or whether the test
itself is actually broken wrt license changes (which would be somewhat
exceptional).  I.e. maybe we can undo the exception after this PR merges?

**Comment by brendandburns** [comment]
@k8s-bot unit test this please

**Comment by karlkfi** [comment]
Sorry I'm late to the party here. 
The license detection uses a third party sentence matching tool. So if there's a file with multiple licenses it can get confused. I tried to make the output as deterministic as possible, but it can still get confused. If it gets confused, just add an exception. The exception can be any text, but there isn't a great way yet to say "multiple". In another case the code was MIT and the tests were BSD in the same package so I made the exception just say MIT. There's no automatic consumer, so just make the exception as clear as possible.

**Comment by karlkfi** [comment]
One thing I did notice recently was that new versions of godep are censoring more license files, like "LICENCE" (uk spelling).  So the vendored files might change without source repo changes.

**Pull request merged**